python/cells.cpp

```diff
@@ -0,0 +1,226 @@
+        .def("__str__",  [](const arb::spike_source_cell&){return "<spike_source_cell>";});
+
+    pybind11::class_<arb::benchmark_cell> benchmark_cell(m, "benchmark_cell",
+        "A benchmarking cell, used by Arbor developers to test communication performance.");
+
+    benchmark_cell
+        .def(pybind11::init<>(
+            [](const regular_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def(pybind11::init<>(
+            [](const explicit_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def("__repr__", [](const arb::spike_source_cell&){return "<benchmark_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<benchmark_cell>";});
+
+    pybind11::class_<arb::lif_cell> lif_cell(m, "lif_cell",
+        "A benchmarking cell, used by Arbor developers to test communication performance.");
+
+    lif_cell
+        .def(pybind11::init<>())
+        .def_readwrite("tau_m", &arb::lif_cell::tau_m,  "Membrane potential decaying constant [ms].")
+        .def_readwrite("V_th",  &arb::lif_cell::V_th,   "Firing threshold [mV].")
+        .def_readwrite("C_m",   &arb::lif_cell::C_m,    "Membrane capacitance [pF].")
+        .def_readwrite("E_L",   &arb::lif_cell::E_L,    "Resting potential [mV].")
+        .def_readwrite("V_m",   &arb::lif_cell::V_m,    "Initial value of the Membrane potential [mV].")
+        .def_readwrite("t_ref", &arb::lif_cell::t_ref,  "Refractory period [ms].")
+        .def_readwrite("V_reset", &arb::lif_cell::V_reset, "Reset potential [mV].")
+        .def("__repr__", [](const arb::lif_cell&){return "<lif_cell>";})
+        .def("__str__",  [](const arb::lif_cell&){return "<lif_cell>";});
+
+    pybind11::class_<cell_parameters> cell_params(m, "cell_parameters", "Parameters used to generate the random cell morphologies.");
+    cell_params
+        .def(pybind11::init<>())
+        .def_readwrite("depth", &cell_parameters::max_depth,"The maximum depth of the branch structure.")
+        .def_readwrite("lengths",   &cell_parameters::lengths,  "Length of branch in μm [range].")
+        .def_readwrite("synapses",  &cell_parameters::synapses, "The number of randomly generated synapses on the cell.")
+        .def_readwrite("branch_probs", &cell_parameters::branch_probs, "Probability of a branch occuring [range].")
+        .def_readwrite("compartments", &cell_parameters::compartments, "Compartment count on a branch [range].")
+        .def("__repr__", util::to_string<cell_parameters>)
+        .def("__str__",  util::to_string<cell_parameters>);
```

Comment:
- Do we want/need defaults for the parameters?

Reply:
Here are some whitespaces misplaced.

Reply:
It uses the defaults for the underlying C++ type.

Reply:
Sorry, I was not clear on this: I meant again a description for the constructor with the defaults as in `(default X)`.

Reply:
If we use the defaults provided by the underlying C++ type, then we have to be careful documenting those values in the Python wrapper, because if that value is changed in the C++, it probably won't get updated in the Python doc strings. We should avoid using default values in the docstrings when the value is easy to check and set using accessors.

Reply:
True.

Reply:
We might want to include the attributes into the con`tor in case the C++ default values change (and users rely on them in their scripts).